src/main/java/com/customemoji/CustomEmojiOverlay.java

```diff
@@ -0,0 +1,226 @@
+        {
+            return null;
+        }
+
+        final FontMetrics fm = graphics.getFontMetrics();
+
+        for (CustomEmojiPlugin.Emoji emoji : emojiSuggestions.values())
+        {
+
+            final String text = emoji.getText();
+
+            // build image component
+            BufferedImage bufferedImage = CustomEmojiPlugin.loadImage(emoji.getFile()).unwrap();
+            bufferedImage = scaleDown(bufferedImage, config.maxImageHeight());
+            ImageComponent imageComponent = new ImageComponent(bufferedImage);
+
+            // build line component
+            LineComponent lineComponent = LineComponent.builder().right(text).build();
+            SplitComponent splitComponent = SplitComponent.builder().first(imageComponent).second(lineComponent).orientation(ComponentOrientation.HORIZONTAL).build();
+
+            panelComponent.getChildren().add(splitComponent);
+        }
+
+        return super.render(graphics);
+    }
+
+
+    public static BufferedImage scaleDown(BufferedImage originalImage, int targetHeight) {
+        int originalWidth = originalImage.getWidth();
+        int originalHeight = originalImage.getHeight();
+
+        // Do not scale if already short enough
+        if (originalHeight <= targetHeight) {
+            return originalImage;
+        }
+
+        // Compute new width while preserving aspect ratio
+        double scaleFactor = (double) targetHeight / originalHeight;
+        int newWidth = (int) Math.round(originalWidth * scaleFactor);
+
+        // Create scaled image
+        BufferedImage scaledImage = new BufferedImage(newWidth, targetHeight, originalImage.getType());
+        Graphics2D graphics = scaledImage.createGraphics();
+        graphics.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BILINEAR);
+        graphics.setRenderingHint(RenderingHints.KEY_RENDERING, RenderingHints.VALUE_RENDER_QUALITY);
+        graphics.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
+        graphics.drawImage(originalImage, 0, 0, newWidth, targetHeight, null);
+        graphics.dispose();
+
+        return scaledImage;
+    }
+
+    private static String extractChatInput(String input)
+    {
+        input = Text.removeTags(input);
+        input = removePlayerName(input);
+        input = removeChatInputAsterisk(input);
+        input = removeBeforeLastSpace(input);
+
+        return input;
+    }
+
+    @NonNull
+    private Map<String, CustomEmojiPlugin.Emoji> getEmojiSuggestions(@NonNull String searchTerm)
+    {
+        Map<String, CustomEmojiPlugin.Emoji> matches = new HashMap<>();
+
+        if (searchTerm.trim().isEmpty() || searchTerm.length() < 3)
+        {
+            return matches;
+        }
+
+        String lowerSearch = searchTerm.toLowerCase();
+
+        for (Map.Entry<String, CustomEmojiPlugin.Emoji> entry : this.plugin.emojis.entrySet())
+        {
+            if (matches.size() >= 10)
```

Comment:
Just realized I forgot to make this into a config setting. Will add that after early feedback